### DISCUSSION TITLE
Should Ecto.Type implementations of `equal?/2` be constant time?

### DISCUSSION THREAD

**Discussion by halostatue** [danielberkompas/cloak_ecto#37] [Q&A] [answered]
I’m not _quite_ sure where the various `equal?/2` implementations are used (on plain or encrypted values), but given what `cloak_ecto` is used for, should they not perhaps be constant time comparisons?

**Comment by danielberkompas** [comment, marked as answer]
@halostatue I'm not an expert on when to use constant time, but I believe that Ecto.Type will call `equal?/2` on the unencrypted values. As a result, I don't think it's important to ensure that this is a constant time comparison.
